src/gt4py/ir/nodes.py

```diff
@@ -708,6 +708,25 @@ def full_interval(cls, order=IterationOrder.PARALLEL):
 
         return interval
 
+    def disjoint_from(self, other: "AxisInterval") -> bool:
+        # This made-up constant must be larger than any LevelMarker.offset used
+        DOMAIN_SIZE: int = 1000
+
+        def get_offset(bound: AxisBound) -> int:
+            return (
+                0 + bound.offset if bound.level == LevelMarker.START else DOMAIN_SIZE + bound.offset
```

Comment:
Kind of nit-picking,  but I would use a more meaningful constant value for comparing values starting from the end:

```suggestion
                0 + bound.offset if bound.level == LevelMarker.START else sys.maxsize + bound.offset
```

Reply:
Can you add to `sys.maxsize` consistently without overflow?